sdk/storage/azure-storage-common/inc/azure/storage/common/crypt.hpp

```diff
@@ -3,17 +3,21 @@
 
 #pragma once
 
+#include <azure/core/base64.hpp>
+
 #include <cstdint>
 #include <string>
 #include <vector>
 
 namespace Azure { namespace Storage {
 
+  namespace Internal {
```

Comment:
Why are we making storage common to wrap the functionality from Azure core? What's the point of it?

The API is already exposed for all services. We shouldn't put it under another API within common

Reply:
This is a convenience overload that happens to accept `std::string` and just an implementation detail to avoid having to re-write all the callers.

The alternative, and correct approach (imo), is to update all callers of `Base64EncodeText` to call the byte vector based public `Base64Encode` API coming from `Azure::Core`.

I will leave it up to the storage folks to remove this, if and when they want to, but it shouldn't be public.

Reply:
Hi @ahsonkhan I was thinking, how about we add two functions to convert between `std::string` and `std::vector<uint8_t>` in `azure/core/strings.hpp`. Maybe we want to put them into `Internal` namespace because we don't want to expose them to end users, they're just for azure sdk developers. These two functions can be super useful. This way, we don't need `Base64EncodeText` anymore, and the code in `sdk/core/azure-core/test/ut/base64.cpp` can also be greatly simplified.

Reply:
That seems reasonable as a subsequent change after the Jan milestone, since it will be non-breaking.

Is it safe to assume that `std::string` in C++ will be UTF-8 encoded bytes and is the reinterpret cast for each string character to `uint8_t` allowed?

Reply:
While i'm thinking the implementation will be as simple as

```
std::string ToString(const std::vector<uint8_t>& data) {
    return std::string(data.begin(), data.end());
}

std::vector<uint8_t> ToBinary(const std::string& data) {
    return std::vector<uint8_t>(data.begin(), data.end());
}
```

Reply:
> While i'm thinking the implementation will be as simple as

That's what I am concerned with.

Where do you expect to use this? Depending on the encoding of the binary data, the string representation could be different. What do you expect converting any arbitrary bytes into `std::string` to mean exactly?

For example, imagine this arbitrary byte vector (which doesn't contain valid UTF-8 characters):
```C++
   vector<uint8_t> vect { 226, 40, 161 }; 
   string someText = ToString(vect); // What is the expectation here?
   cout << "Some output: " << someText << endl; // Some non-printable output: �(�
```

Reply:
never mind, I didn't think it through.